### PR TITLE
Add data input distribution for model+data parallel.

### PR DESCRIPTION
**Pull request by qlzh727**
This will be the final change for multi-worker support of data+model distribution.

See the multi-process testing in cl/578919156

### CODE REVIEW

**Comment by qlzh727** [comment]
This should address  and https://github.com/keras-team/keras/issues/18560 and https://github.com/keras-team/keras/issues/18561

**Comment by qlzh727** [comment]
The numpy test failure seems to be unrelated to my change.

File: keras/backend/jax/distribution_lib.py [hunk 1]
```diff
@@ -125,22 +124,53 @@ def distribute_data_input(inputs, layout):
...
-    if per_process_batch_size % per_replica_batch_size != 0:
+    if mesh_rank == 1:
+        # This is data parallel mesh only. We will split the full data
+        # across the batch dim.
+        num_split = jax.local_device_count()
+        per_replica_batch_size = per_process_batch_size // num_split
+        if per_process_batch_size % per_replica_batch_size != 0:
+            raise ValueError(
+                f"The local batch size {per_process_batch_size} is not"
+                "divisible by the number of local replica "
```
Comment (fchollet) [inline keras/backend/jax/distribution_lib.py:140]: replicas (plural)

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**